src/common/snippets/src/pass/align_element_types.cpp

```diff
@@ -0,0 +1,73 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "snippets/pass/align_element_types.hpp"
+#include "snippets/itt.hpp"
+
+namespace ov {
+namespace snippets {
+
+pass::AlignElementTypes::AlignElementTypes(std::vector<ov::element::Type> input_precisions,
+                                           std::vector<ov::element::Type> output_precisions) :
+                                           m_input_precisions(std::move(input_precisions)),
+                                           m_output_precisions(std::move(output_precisions)) {
+}
+
+bool pass::AlignElementTypes::run_on_model(const std::shared_ptr<ov::Model>& m) {
```

Comment:
Could you tell me please if you specially removed [case](https://github.com/openvinotoolkit/openvino/blob/ff991f54b1f5ea36b4a186e715c85306a795f4e8/src/common/snippets/src/op/subgraph.cpp#L553) with `Transpose` or accidentally?

Reply:
No I didn't, it was just not there when the PR was created, and then the changes were ignored during rebase since the whole function was deleted. It's a huge disadvantage of old big PR's. Hope we can merge the last two asap, or similar cases might appear.
Anyway, thank you very much for noticing. I'll replicate the logic from the updated pass